### ISSUE TITLE
docker-compose on a mac became terrifically slow

### ISSUE THREAD

**Issue by mvasin** [docker/compose#4752]
Suddenly, `docker-compose` became terrifically slow. I'm eager to find out why.

Simple `docker-compose ps` when there are no running containers takes 10 seconds:

```
$ time docker-compose ps
Name   Command   State   Ports
------------------------------

real	0m10.320s
user	0m0.232s
sys	0m0.061s
```

while `docker`'s `ps` takes 0.02 seconds:
```
$ time docker ps
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES

real	0m0.019s
user	0m0.008s
sys	0m0.006s
```

Getting `up` this `docker-compose.yml` (with image already pulled in)
```
version: "3"
services:
  ubuntu:
    image: ubuntu
    command: ["echo", "hello"]
```

takes more then 1 minute 16 seconds:
```
$ time docker-compose up -d
Creating network "test_default" with the default driver
Creating test_ubuntu_1

real	1m16.356s
user	0m0.459s
sys	0m0.112s
```
(I'm not sure why I don't see 'hello' output here, tell me if you know)

while `docker` does the same in less then a second:
```
$ time docker run ubuntu echo hello
hello

real	0m0.794s
user	0m0.011s
sys	0m0.010s
```

I'm running docker version 17.03.1-ce, build c6d412e and docker-compose version 1.11.2, build dfed245 on macOS 10.12.4.

**Comment by mvasin** [comment]
Reinstalling the whole macOS didn't help, it is still sluggish.

**Comment by shin-** [comment]
Can you try with `--verbose` and see if it hangs at a particular point?

You may also want to upgrade to 1.12.0 and see if it changes anything.

**Comment by ladyrassilon** [comment]
I'm finding it hanging on doing docker-compose build for minutes, if not longer.

Its not related to the machine, (its happening on native, virtualbox and aws), with both docker for mac (compose 1.12) and docker-compose from homebrew.

**Comment by shin-** [comment]
@ladyrassilon Is it slow with any project? What happens if you run `build` in an barebones project folder (just a Dockerfile and `docker-compose.yml`)?

**Comment by aleCeres** [comment]
same issue...so slow

**Comment by mvasin** [comment]
This trick seems to heal it: https://github.com/docker/compose/issues/3419#issuecomment-221793401

**Comment by ladyrassilon** [comment]
@shin- I've diagnosed it as the time it takes to send context to the docker demon, which is reported when you do a manual docker build, but not when you do a docker-compose frustratingly which is rather usability compromised.... 

A progress bar would be helpful.

**Comment by hughbrien** [comment]
I am having the same problem, which started around Jun 16.  docker-compose has just become slow.  I have been messing around with kubernetes implementations on the my Mac so I think it has something to do with that.  I am going to try to try:  #3419 (comment)

**Comment by macropin** [comment]
I hit this suddenly after changing from ethernet to wifi. As above https://github.com/docker/compose/issues/3419#issuecomment-221793401 seems to fix the issue

**Comment by PEZO19** [comment]
Still exists for me, and my /etc/hosts gets overwritten again and again... Do you know how to fix it permantently?

**Comment by shin-** [comment]
See ongoing discussion on #3419

**Comment by hrajchert** [comment]
For me the comment that solved the problem was this one: https://github.com/docker/compose/issues/3419#issuecomment-230827741

**Comment by phazei** [comment]
For me it fixed it with #3419 but with 'localunixsocket.localdomain' which I found through the tcpdump cmd listed

**Comment by marcelfalliere** [comment]
`docker-compose version 1.22.0, build f46880f`

````
$> NO_PROXY=* docker-compose --verbose -f docker-compose.dev.yml up --build
...
compose.service.build: Building patients.api
compose.cli.verbose_proxy.proxy_callable: docker build <- (path='/Users/marcelfalliere/IntactCode/myProject-v2/cswebpatientserver', tag='myProject/cswebpatientserver_dev.api:latest', rm=True, forcerm=False, pull=False, nocache=False, dockerfile='Dockerfile.dev', cache_from=None, labels=None, buildargs={}, network_mode=None, target=None, shmsize=None, extra_hosts=None, container_limits={'memory': None}, gzip=False, isolation=None, platform=None)
`````

it hangs there for **two minutes or more** ... then :


````
docker.api.build._set_auth_headers: Looking for auth config
docker.auth.resolve_authconfig: Using credentials store "osxkeychain"
docker.auth._resolve_authconfig_credstore: Looking for auth entry for 'myProjectContainerRegistry.azurecr.io'
docker.auth.resolve_authconfig: Using credentials store "osxkeychain"
docker.auth._resolve_authconfig_credstore: Looking for auth entry for 'https://myProjectcontainerregistry.azurecr.io'
docker.auth.resolve_authconfig: Using credentials store "osxkeychain"
docker.auth._resolve_authconfig_credstore: Looking for auth entry for 'https://myProjectContainerRegistry.azurecr.io'
docker.auth.resolve_authconfig: Using credentials store "osxkeychain"
````

and proceed to build the image, using cache.

I have added these three line in my `/etc/hosts` :

````
127.0.0.1   localunixsocket.local
127.0.0.1   localunixsocket.lan
127.0.0.1   localunixsocket
````

The `tcpdump` trick to watch what's going does not yield stuff reltive to docker (docs.google, creative cloud etc...)

It seems to be looking at credentials, but this docker compose I'm building does not use a repository. It seems to be trying to connect to our container registry on Azure. 

Any hints at what's actually going on ?
